modules/resolve.js

```diff
@@ -0,0 +1,83 @@
+export const resolveLocation = (location, base) => {
+  if (typeof location === 'string') {
+    return resolve(location, base)
+  } else {
+    location.pathname = resolve(location.pathname, base)
```

Comment:
I tested out `resolve-pathname` by adding a `noncompliant` mode to it: `resolvePathname(to, from, noncompliant)`. Essentially all that it does is to not pop the filename (https://github.com/mjackson/resolve-pathname/blob/master/modules/index.js#L26) when in that mode. I also added the modified examples from RFC 3986 to test this out. There were a couple of issues:

1. It does not work when `to` includes a hash or search string. I would expect `resolvePathname('?x=y', '/a/b', true)` to resolve to `/a/b?x=y`, but it is currently including a trailing slash and resolving to `/a/b/?x=y`. This also fails for abnormal examples like `g#s/../x` resolving to `/a/b/c/d/x`. I was dealing with that by checking for the earliest index of `?` or `#`, stripping that index to the end of of the string off, resolving, then appending that to the resolved pathname before returning. It is easy to add that functionality to `resolvePathname` assuming you want to support that.

2. There are some dot and double-dot notation issues. It gets a little bit complicated because this isn't RFC compliant, but I feel that `resolvePathname('..', '/a/b/c/d', true)` should resolve to `/a/b/c` while it is resolving to `/a/b/c/`.